server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -349,9 +350,17 @@ public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower
                 throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() +
                         "] does not support DISJOINT ranges");
             }
-            DateMathParser parser = forcedDateParser == null
-                    ? dateMathParser
-                    : forcedDateParser;
+            DateMathParser parser;
+            if (forcedDateParser == null) {
+                if (lowerTerm instanceof Integer || upperTerm instanceof Integer) {
+                    // force epoch_millis
+                    parser = EPOCH_MILLIS_PARSER;
```

Comment:
Should we just cast them to a number and call `longValue` on them? It feels like a shame to `toString` them and then parse them if we know they are numbers.